fix(ble): fix race condition for MAC addr device ID in sample

gminn opened the pull request:
### Summary
  The device ID when using the BLE MAC address is `Unknown` after the system
  boots because the device has not been given a MAC address yet at the
  time the device ID is set (system init). This change fixes that by setting a
  static device id for the `peripheral_mds` sample app.

 Bonus - enabled logging at the `INFO` level on the `peripheral_mds`
 app so we can run commands like `mflt get_device_info` and get print-outs.

  ### Test Plan
  - Tested on a nRF52840 DK with the peripheral_mds sample app
  ```
uart:~$ *** Booting nRF Connect SDK v3.5.99-ncs1-4965-g3733e7097909 ***
Starting Bluetooth Memfault example
Bluetooth initialized
Advertising successfully started
uart:~$ mflt get_device_info
[00:00:03.995,544] <inf> mflt: S/N: nrf-ble-testdevice
[00:00:03.995,605] <inf> mflt: SW type: nrf-ble-fw
[00:00:03.995,666] <inf> mflt: SW version: 0.0.1+688c47
[00:00:03.995,727] <inf> mflt: HW version: nrf52840dk_nrf52840
  ```

---

Resolves: MCU-433

gminn commented:
* **#24** <a href="https://app.graphite.dev/github/pr/memfault/sdk-nrf/24?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gminn and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

ejohnso49 reviewed (commented):
Thought about this more, and I think I've sadly come to the conclusion that we just shouldn't try to use this for the BLE MAC case. I think that IMEI and network MAC are fine to keep in. Maybe we add a little blurb to the Kconfig to indicate this is for a network MAC (ie nrf7002).

Also I realized that the device serial isn't needed for reboot events 😑 should have remembered that. There's a Memfault config to include device serial with events but it's disabled by default because nearly all devices use the device serial in the chunks POST url.

gminn commented:
> Thought about this more, and I think I've sadly come to the conclusion that we just shouldn't try to use this for the BLE MAC case.

To clarify -- you're thinking we remove the example of using the BLE MAC addr for the device id?

gminn committed:
fix(ble): set device ID after BLE init when using MAC address

  ### Summary
  The device ID when using the MAC address was Unknown after the system
  booted because the device has not been given a MAC address yet at the
  time the device ID is set (system init). This change removes the option
  for using a MAC address as the device id for BLE devices, and sets a
  static device id for the BLE sample app.

  ### Test Plan
  - Tested on a nRF52840 DK with the peripheral_mds sample app
  ```
uart:~$ *** Booting nRF Connect SDK v3.5.99-ncs1-4965-g3733e7097909 ***
Starting Bluetooth Memfault example
Bluetooth initialized
Advertising successfully started
uart:~$ mflt get_device_info
[00:00:03.995,544] <inf> mflt: S/N: nrf-ble-testdevice
[00:00:03.995,605] <inf> mflt: SW type: nrf-ble-fw
[00:00:03.995,666] <inf> mflt: SW version: 0.0.1+688c47
[00:00:03.995,727] <inf> mflt: HW version: nrf52840dk_nrf52840
  ```

gminn commented:
> To clarify -- you're thinking we remove the example of using the BLE MAC addr for the device id?

@ejohnso49 This is back on my radar and I applied your suggestions -- let me know what you think!

ejohnso49 commented on modules/memfault-firmware-sdk/Kconfig:
```diff
@@ -48,6 +48,7 @@ config MEMFAULT_NCS_DEVICE_ID_IMEI
 config MEMFAULT_NCS_DEVICE_ID_NET_MAC
 	bool "Use network interface MAC address as device ID"
 	select HW_ID_LIBRARY
+	depends on !BT
```
I think this might need to be removed because a 7002 application with an nRF53 would still use the BLE stack. I think your other changes cover things as best as we can

gminn replied:
Ah yes good point. I double checked after removing this that we get a proper MAC address device id 👍

```
uart:~$ mflt get_device_info
[00:00:12.471,466] <inf> mflt: S/N: F4CE3600260E
[00:00:12.471,496] <inf> mflt: SW type: nrf7002_nrf53-fw
[00:00:12.471,527] <inf> mflt: SW version: 0.0.1+cf5686
[00:00:12.471,557] <inf> mflt: HW version: nrf7002dk
```

added to the `samples/wifi/shell` app's `prj.conf` to get that build:
```
CONFIG_MEMFAULT=y
CONFIG_MEMFAULT_SHELL=y
CONFIG_MEMFAULT_LOGGING_ENABLE=y
CONFIG_MEMFAULT_LOG_LEVEL_INF=y

CONFIG_BT=y
CONFIG_BT_SMP=y
CONFIG_BT_PERIPHERAL=y
CONFIG_BT_DEVICE_NAME="Nordic_Memfault"
CONFIG_BT_PRIVACY=y
```

and then built with
```
west build --pristine always --board nrf7002dk/nrf5340/cpuapp nrf/samples/wifi/shell -- -DCONFIG_MEMFAULT_NCS_PROJECT_KEY=\"dummy\"
```

ejohnso49 reviewed (approved):
Thanks for tackling this, at least we won't have blank device serial's by default

gminn reviewed (commented):
Merged upstream in https://github.com/nrfconnect/sdk-nrf/commit/f3f5ff68b0ca5d680dc4ad31b6304f48eb61449e !
